### PR TITLE
Fix ROCm conformance tests

### PR DESCRIPTION
**Pull request by sogartar**
This PR requires to expose additional compiler arguments with HAL conformance test that require compilation of MLIR.
I added them here. I can put them in a separate PR.

### CODE REVIEW

File: build_tools/cmake/iree_hal_cts_test_suite.cmake [hunk 1]
```diff
@@ -23,6 +23,8 @@ include(CMakeParseArguments)
 #       `-iree-hal-target-backends` option of `iree-compile` to use for
 #       executable generation. If this is omitted, or the associated compiler
 #       target is not enabled, tests which use executables will be disabled.
+#   COMPILER_ARGS: Additional compiler arguments.
```
Comment (benvanik) [inline build_tools/cmake/iree_hal_cts_test_suite.cmake]: COMPILER_FLAGS for consistency with other rules

File: experimental/rocm/native_executable.c [hunk 2]
```diff
@@ -74,22 +74,30 @@ iree_status_t iree_hal_rocm_native_executable_create(
...
   iree_status_t status = iree_allocator_malloc(context->host_allocator,
                                                total_size, (void**)&executable);
-  executable->pipeline_layouts =
-      (void*)((char*)executable + sizeof(*executable) +
-              entry_count * sizeof(iree_hal_rocm_native_executable_function_t));
-  hipModule_t module = NULL;
   if (iree_status_is_ok(status)) {
+    executable->context = context;
+    iree_hal_resource_initialize(&iree_hal_rocm_native_executable_vtable,
+                                 &executable->resource);
```
Comment (benvanik) [inline experimental/rocm/native_executable.c]: iree_hal_resource_initialize usually goes first
```suggestion
    iree_hal_resource_initialize(&iree_hal_rocm_native_executable_vtable,
                                 &executable->resource);
    executable->context = context;
```

File: build_tools/cmake/iree_hal_cts_test_suite.cmake (hunk 1 above)
Comment (sogartar) [reply build_tools/cmake/iree_hal_cts_test_suite.cmake]: I renamed it.

File: experimental/rocm/native_executable.c (hunk 2 above)
Comment (sogartar) [reply experimental/rocm/native_executable.c]: I changed the order.

**Comment by sogartar** [comment]
After rebasing there is a new `file_test` conformance test that is now failing. Not to leave this PR hanging I will fix this test in another PR.

**Comment by sogartar** [comment]
This PR may not be squashed when merging as commits are logically separated.

**Comment by sogartar** [comment]
@benvanik did you forget to approve this?

**Review by benvanik — APPROVED** [review]
yes :)

**Comment by sogartar** [comment]
There was some error with the auto-merge. Locally I am able to rebase it without conflicts. Could someone with access merge it?

**Comment by ScottTodd** [comment]
> This PR may not be squashed when merging as commits are logically separated.

Which merge strategy would you like? 
image

Seems like a good case for rebase and merge?

**Comment by ScottTodd** [comment]
Or create a merge commit?

**Comment by sogartar** [comment]
> Or create a merge commit?

They can be left as is since they are logically separated. Also it is better to not mix the experimental ROCm driver stuff with the rest.

**Comment by ScottTodd** [comment]
Used rebase and merge, since a merge commit would be `--no-ff`, and I couldn't cleanly `--ff-only` locally. Ehh.. hope that looks okay in the history.

**Pull request merged**